### DISCUSSION TITLE
GitHub pages

### DISCUSSION THREAD

**Discussion by olzama** [delph-in/docs#79] [Team Posts]
I've been trying to experiment with GitHub pages as well, to see if we can use the docs repo to project a convenient portal, e.g. for the current summit or for the organization in general? Since the wiki is not the best place to look at a summit as an event (the structure ends up being less consistent). However I cannot figure out how to make it actually read in all the current wiki pages. I can just link to them directly from the website, which kind of defeats the purpose, in the sense that then the website can be anywhere.

[https://delph-in.github.io/docs/index.html](https://delph-in.github.io/docs/index.html)

-- and we could add good organization for the summit here, for example, provided we can directly use the wiki pages. I mean we can always just link to them. But then maybe the website should not be published from the docs repo, rather we could have a website for the summits, highlighting the current one, if we so choose (I personally think it would be nice). 

Anyway, I have a sense that there is some fruit to be reaped from this GitHub pages thing in combination with the wiki but I do not yet understand how exactly :).

**Comment by goodmami** [comment]
Do you want to duplicate the wiki in GitHub pages? I don't think that's a great idea, although previously I suggested that we could use *only* GitHub pages for our "wiki" (since you can edit files on GitHub much like a wiki) so that we could mirror the URL structure from the delph-in.net Moin wikis. It was also suggested that we publish our semi-official documentation (such as the current RFC pages) on GitHub Pages instead of a wiki.

Anyway, with GitHub pages you'll get the ability to do custom CSS or javascript, and you'll have potentially shorter URLs. If we want to keep managing summits in a wiki, I think we should just stick to the wiki. Otherwise we could, for instance, make a new repository for each summit so the URL could be something like https://delph-in.github.io/2021.

If you want a GitHub Pages landing site for the organization, you'd create a repository called `delph-in.github.io` and publish from there.

**Comment by olzama** [comment]
Nah, I meant, project, not duplicate. So, the content is added via the wiki but there is a website with a better structure which displays that content. But I think I am imagining things and there is no way of doing that maybe.

I thought about just using repos for each summit, too, but github pages sites on their own don't appear to have search. You have to implement it on the client site, and I looked at it and thought it looked brittle for us to rely on as an organization. And the wiki does have great search (and that seems even more important than good structure).

**Comment by goodmami** [comment]
If you're sure you won't use GitHub pages for the `docs` repo for anything else, you can probably setup an action workflow that pulls the wiki and publishes it to the `gh-pages` branch. There are ways of setting up client-side search, like is done with Sphinx documentation (e.g., the search in [PyDelphin's documentation](https://pydelphin.readthedocs.io/en/latest/)). I'm not sure it's worth the effort, though :P

**Comment by olzama** [comment]
> There are ways of setting up client-side search, like is done with Sphinx documentation (e.g., the search in PyDelphin's documentation). I'm not sure it's worth the effort, though :P

exactly :) that's what I thought when I looked into it. Sounds brittle. But if at some point they integrate a GH-supported search, then we may look into it again, in hopes of having better portals/organization for summit info.

**Comment by arademaker** [comment]
> It was also suggested that we publish our semi-official documentation (such as the current RFC pages) on GitHub Pages instead of a wiki.

I have also suggest it for contents with a well defined structure such as ErgSemantics.  I believe it makes sense to have these contents in the docs repository.

**Comment by arademaker** [comment]
I believe an repository to each summit is too much. I would vote and have a github pages for the docs repository and move gradually to it the more stable contents from the wiki. Or contents with a well defined structure like pages of the summits (home, program, info about trip, participants etc. discussions and transcriptions can stay in the wiki)! I will actually have few pages to rename to improve consistency.

**Comment by arademaker** [comment]
> Nah, I meant, project, not duplicate. So, the content is added via the wiki but there is a website with a better structure which displays that content. But I think I am imagining things and there is no way of doing that maybe.

It is possible. We can have in the website links to pages we can create in advance for the summit. The discussions, once defined can have the pages created, links in the website added and just content let to be filled later or during the meeting

**Comment by goodmami** [comment]
> It is possible.

Whatever solution we find, I hope it is as simple as possible. I don't think future maintainers of this space should be required to know the ins and outs of publishing on GitHub. Hopefully they can just create or edit a new markdown file and thereby edit a page that gets rendered and published automatically. And this process should have some minimal documentation.

**Comment by arademaker** [comment]
I agree, as I said before, my reference is how http://universaldependencies.org is doing. It seems to work fine and the content is based on http://github.com/universaldependencies/docs.
